Extensions/FileSystem/JsExtension.js

```diff
@@ -59,6 +59,77 @@ module.exports = {
       )
       .setFunctionName("gdjs.fileSystem.makeDirectory");
 
+    extension
+      .addAction(
+        "SaveStringToFile",
+        t("Save a text into a file"),
+        t("Save a text into a file."),
+        t("Save _PARAM0_ into file _PARAM1_"),
+        t("Filesystem/Windows, Linux, MacOS"),
+        "JsPlatform/Extensions/filesystem_save_file24.png",
+        "JsPlatform/Extensions/filesystem_save_file32.png"
+      )
+      .addParameter("string", t("String (text)"), "", false)
+      .addParameter("string", t("Save path"), "", false)
+      .getCodeExtraInformation()
+      .setIncludeFile(
+        "Extensions/FileSystem/filesystemtools.js"
+      )
+      .setFunctionName("gdjs.fileSystem.saveStringToFile");
+
+    extension
+      .addAction(
+        "LoadStringFromFileAsync",
+        t("Load a text from a file (Async)"),
+        t("Load a text from a file asynchronously. Use this for large files to not delay game execution. (The content of the file will not be available immediately)"),
+        t("Load text from _PARAM1_ into scene variable _PARAM0_ (Async)"),
+        t("Filesystem/Windows, Linux, MacOS"),
+        "JsPlatform/Extensions/filesystem_load_file24.png",
+        "JsPlatform/Extensions/filesystem_load_file32.png"
+      )
+      .addParameter("scenevar", t("Scene variable"), "", false)
+      .addParameter("string", t("Load path"), "", false)
+      .getCodeExtraInformation()
+      .setIncludeFile(
+        "Extensions/FileSystem/filesystemtools.js"
+      )
+      .setFunctionName("gdjs.fileSystem.loadStringFromFileAsync");
+
+    extension
+      .addAction(
+        "LoadStringFromFileSync",
+        t("Load a text from a file"),
+        t("Load a text from a file. Only use this on small files to not delay the game execution."),
```

Comment:
I would say "to avoid any lag or freeze during the the game execution." (which is I think a bit more clear than "delaying game execution").